src/vs/workbench/contrib/files/browser/views/explorerViewer.ts

```diff
@@ -1346,36 +1355,58 @@ export class FileDragAndDrop implements ITreeDragAndDrop<ExplorerItem> {
 		return this.workspaceEditingService.updateFolders(0, workspaceCreationData.length, workspaceCreationData);
 	}
 
-	private async doHandleExplorerDrop(source: ExplorerItem, target: ExplorerItem, isCopy: boolean): Promise<void> {
+	private async doHandleExplorerDrop(sources: ExplorerItem[], target: ExplorerItem, isCopy: boolean): Promise<void> {
 		// Reuse duplicate action if user copies
+		const sourceTargetPairs: { source: URI; target: URI }[] = [];
 		if (isCopy) {
 			const incrementalNaming = this.configurationService.getValue<IFilesConfiguration>().explorer.incrementalNaming;
-			const stat = await this.workingCopyFileService.copy(source.resource, findValidPasteFileTarget(this.explorerService, target, { resource: source.resource, isDirectory: source.isDirectory, allowOverwrite: false }, incrementalNaming));
-			if (!stat.isDirectory) {
-				await this.editorService.openEditor({ resource: stat.resource, options: { pinned: true } });
+			for (const source of sources) {
+				sourceTargetPairs.push({ source: source.resource, target: findValidPasteFileTarget(this.explorerService, target, { resource: source.resource, isDirectory: source.isDirectory, allowOverwrite: false }, incrementalNaming) });
 			}
+			const stats = await this.workingCopyFileService.copy(sourceTargetPairs);
+			stats.forEach(async stat => {
+				if (!stat.isDirectory) {
+					await this.editorService.openEditor({ resource: stat.resource, options: { pinned: true } });
```

Comment:
You open editors one by one when copied, you should use the openEditors method to bulk open imho

Reply:
I'll look into this also :+1: